components/feature/readerview/src/main/java/mozilla/components/feature/readerview/ReaderViewFeature.kt

```diff
@@ -0,0 +1,46 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.feature.readerview
+
+import android.support.annotation.VisibleForTesting
+import mozilla.components.browser.session.SelectionAwareSessionObserver
+import mozilla.components.browser.session.SessionManager
+import mozilla.components.concept.engine.Engine
+import mozilla.components.concept.engine.webextension.WebExtension
+import mozilla.components.concept.toolbar.Toolbar
+import mozilla.components.support.base.feature.LifecycleAwareFeature
+import mozilla.components.support.base.log.logger.Logger
+
+class ReaderViewFeature(
+    private val engine: Engine,
+    private val toolbar: Toolbar,
+    sessionManager: SessionManager
+) : SelectionAwareSessionObserver(sessionManager), LifecycleAwareFeature {
+
+    @VisibleForTesting
+    internal var readerViewWebExt = WebExtension("mozac-readerview", "resource://android/assets/extensions/readerview/")
+    internal var installed = false
+
+    override fun start() {
+        // TODO Create feature-webextension to manage WebExtensions centrally?
+        if (!installed) {
+            engine.installWebExtension(readerViewWebExt,
+                onSuccess = {
+                    Logger.debug("Installed browser-icons extension")
+                    installed = true
+                },
+                onError = { ext, throwable ->
+                    Logger.error("Failed to install extension ${ext.id}", throwable)
+                }
+            )
+        }
```

Comment:
Since this feature has activity/fragment scope and the engine "global" scope: What happens if this feature gets initialized multiple times (e.g. activity re-creation) and we install the extension again? Maybe that's another reason for separate features - a global one (that sets everything up) and a "local" one for messing with the UI?

Reply:
yeah the web extension life cycle is tied to the runtime not the session.

Reply:
True, there's already a check that makes sure we only install once, but we can split things up when we bring it all the pieces together.